### PR TITLE
Expose the exportLinks object to FileData

### PR DESCRIPTION
**Pull request by league**
I'm using this to access Google Spreadsheets, which don't seem to provide a normal `fileDownloadUrl`. Instead, the API provides "exportLinks" key, which is a map from MIME type to a download URL. This small patch just makes the export links accessible in the `FileData` structure.

An example of exportLinks (JSON notation):

```
{"text/csv":
   "https://docs.google.com/sheets/export?id=XYZ&exportFormat=csv",
 "application/pdf":
   "https://docs.google.com/sheets/export?id=XYZ&exportFormat=pdf",
 "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet":
   "https://docs.google.com/spreadsheets/export?id=XYZ&exportFormat=xlsx"
 }
```

### CODE REVIEW

File: src/Network/Google/Drive/File.hs [hunk 1]
```diff
@@ -87,6 +91,7 @@ instance FromJSON FileData where
         <*> (fmap read <$> o .:? "fileSize") -- fileSize is a String!
         <*> o .:? "downloadUrl"
         <*> o .: "mimeType"
+        <*> (fromMaybe empty <$> o .:? "exportLinks")
```
Comment (pbrisbin) [inline src/Network/Google/Drive/File.hs]: How about `o .:? "exportLinks" .!= empty`?

File: src/Network/Google/Drive/File.hs [hunk 2]
```diff
@@ -42,7 +43,8 @@ import Network.Google.Api
 import Control.Applicative ((<$>), (<*>))
 import Control.Monad (mzero, void)
 import Data.Aeson
-import Data.Maybe (isJust)
+import Data.HashMap.Strict (HashMap, empty)
+import Data.Maybe (isJust, fromMaybe)
```
Comment (pbrisbin) [inline src/Network/Google/Drive/File.hs]: Reminder that if you use `(.!=)`, you won't need to import `fromMaybe` anymore.

**Comment by pbrisbin** [comment]
Minor comment, then I'll be happy to merge. Thanks!

File: src/Network/Google/Drive/File.hs [hunk 3]
```diff
@@ -64,6 +67,7 @@ data FileData = FileData
     , fileSize :: !(Maybe Int)
     , fileDownloadUrl :: !(Maybe Text)
     , fileMimeType :: !Text
```
Comment (pbrisbin) [inline src/Network/Google/Drive/File.hs]: I would re-use the type synonym you've added here (and anywhere else `fileMimeType` is used).

**Comment by league** [comment]
Great suggestions – I hadn't bothered to understand `(.!=)` until now. Thanks.

**Pull request merged**